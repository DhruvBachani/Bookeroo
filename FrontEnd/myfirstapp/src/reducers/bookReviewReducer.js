import { GET_REVIEWS, GET_USER } from "../actions/types";

export const initialState = {
  bookReviews: []
};

export default function(state = initialState, action) {
  switch (action.type) {
    case GET_REVIEWS:
      console.log(action.payload);
      return {
        ...state,
        bookReviews: action.payload
      };

    default:
      return state;
  }
}
