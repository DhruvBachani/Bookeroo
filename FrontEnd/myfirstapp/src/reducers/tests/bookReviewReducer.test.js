import React from 'react';
import reducer from '../../../src/reducers'
import { shallow } from 'enzyme';
import { initialState } from '../bookReviewReducer';

describe('bookReviewReducer', () => {
    it('should return the initial state', () => {
        expect(reducer(undefined, {})).toMatchSnapshot();
    })

    it('should handle GET_REVIEWS', () => {
        expect(
            reducer(initialState,
                {
                    type: 'GET_REVIEWS'
                })
        ).toMatchSnapshot();
    })
})