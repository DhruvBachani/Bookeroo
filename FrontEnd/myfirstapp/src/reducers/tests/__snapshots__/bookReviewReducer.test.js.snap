// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bookReviewReducer should handle GET_REVIEWS 1`] = `
Object {
  "book": Object {
    "allCategories": Array [],
    "book": Array [],
    "books": Array [],
  },
  "errors": Object {},
  "person": Object {
    "person": Object {},
    "persons": Array [],
  },
  "requests": Object {
    "userRequests": Array [],
  },
  "review": Object {
    "bookReviews": undefined,
  },
  "security": Object {
    "user": Object {},
    "validToken": false,
  },
}
`;

exports[`bookReviewReducer should return the initial state 1`] = `
Object {
  "book": Object {
    "allCategories": Array [],
    "book": Array [],
    "books": Array [],
  },
  "errors": Object {},
  "person": Object {
    "person": Object {},
    "persons": Array [],
  },
  "requests": Object {
    "userRequests": Array [],
  },
  "review": Object {
    "bookReviews": Array [],
  },
  "security": Object {
    "user": Object {},
    "validToken": false,
  },
}
`;
